File: sdk/python/printer_client.py
import requests
import base64
import os

class PrinterGateway:
    """SDK Python Simplificado para Printer API Gateway."""
    
    def __init__(self, base_url="http://localhost:5000", api_key="minha_chave_segura_123"):
        self.base_url = base_url.rstrip("/")
        self.headers = {"X-Api-Key": api_key, "Content-Type": "application/json"}

    def imprimir(self, num_pedido: str, impressora: str, conteudo: str):
        """
        Envia um trabalho de impressão. 
        Detecta automaticamente se o conteúdo é PDF (Base64 ou Caminho), ZPL ou Texto.
        """
        # Se for um caminho de arquivo PDF, converte para Base64 automaticamente
        if conteudo.lower().endswith(".pdf") and os.path.exists(conteudo):
            with open(conteudo, "rb") as f:
                conteudo = base64.b64encode(f.read()).decode("utf-8")
        
        payload = {
            "num_pedido": num_pedido,
            "impressora": impressora,
            "conteudo": conteudo
        }
        
        response = requests.post(f"{self.base_url}/imprimir", headers=self.headers, json=payload)
        response.raise_for_status()
        return response.json()

    def listar_impressoras(self):
        """Retorna o status atual de todas as impressoras."""
        response = requests.get(f"{self.base_url}/impressoras", headers=self.headers)
        response.raise_for_status()
        return response.json()

File: sdk/typescript/PrinterClient.ts
/**
 * Printer API Gateway - Simplified TypeScript SDK
 */

export interface PrintRequest {
  num_pedido: string;
  impressora: string;
  conteudo: string;
}

export interface PrintResponse {
  status: string;
  num_pedido: string;
  tipo_detectado: string;
}

export class PrinterGateway {
  private baseUrl: string;
  private apiKey: string;

  constructor(baseUrl: string = 'http://localhost:5000', apiKey: string = 'minha_chave_segura_123') {
    this.baseUrl = baseUrl.replace(/\/$/, '');
    this.apiKey = apiKey;
  }

  private get headers(): HeadersInit {
    return {
      'X-Api-Key': this.apiKey,
      'Content-Type': 'application/json',
    };
  }

  /**
   * Envia um trabalho de impressão (Auto-detecta PDF, ZPL ou Texto).
   */
  async imprimir(num_pedido: string, impressora: string, conteudo: string): Promise<PrintResponse> {
    const response = await fetch(`${this.baseUrl}/imprimir`, {
      method: 'POST',
      headers: this.headers,
      body: JSON.stringify({ num_pedido, impressora, conteudo }),
    });

    if (!response.ok) {
      const errorData = await response.json();
      throw new Error(errorData.detail || 'Erro na API de impressão');
    }

    return await response.json();
  }

  /**
   * Lista todas as impressoras e seus status.
   */
  async listarImpressoras(): Promise<any> {
    const response = await fetch(`${this.baseUrl}/impressoras`, {
      method: 'GET',
      headers: this.headers,
    });
    if (!response.ok) throw new Error('Falha ao listar impressoras');
    return await response.json();
  }
}
